packages/DataGrid/src/DataGrid.js

```diff
@@ -0,0 +1,628 @@
+    if (data.length && pageSize === null) {
+      setPageSize(() => data.length);
+    }
+  }, [data.length, pageSize]);
+
+  React.useEffect(() => {
+    refPrevLastScrollHeight.current = refScrollGrid.current.scrollHeight;
+  }, []);
+
+  React.useEffect(() => {
+    // this is required to readjust the active highlight
+    // after any rerender
+    if (refPrevLastScrollHeight.current && refPrevLastScrollHeight.current < refScrollGrid.current.scrollHeight) {
+      refScrollGrid.current.scrollTo(0, refScrollGrid.current.scrollTop + 1);
+    }
+    restoreHighlightFocus();
+  });
+
+  React.useImperativeHandle(
+    ref,
+    () => {
+      return {
+        focus: () => {
+          restoreHighlightFocus();
+        },
+        getVisibleIndexes: () => {
+          return refVisibleIndexes.current;
+        },
+        getVisibleRows: () => {
+          const { start, stop } = refVisibleIndexes.current;
+          return data.slice(start, stop);
+        },
+      };
+    },
+    [data, restoreHighlightFocus]
+  );
+
+  const handleBlurGrid = React.useCallback(() => {
+    const $isActive = refContainer.current.querySelector(".grid--is-active");
+    if ($isActive) $isActive.classList.toggle("grid--is-blur");
+  }, []);
+
+  const handleMouseUpGrid = React.useCallback(
+    event => {
+      handleClick({ data, ColumnDefinitions })(event);
+    },
+    [ColumnDefinitions, data, handleClick]
+  );
+
+  const handleKeyDownGrid = React.useCallback(
+    event => {
+      handleKeyDown({ data, ColumnDefinitions })(event);
+    },
+    [ColumnDefinitions, data, handleKeyDown]
+  );
+
+  const handleKeyUpGrid = React.useCallback(
+    event => {
+      handleKeyUp({ data, ColumnDefinitions })(event);
+    },
+    [ColumnDefinitions, data, handleKeyUp]
+  );
+
+  const handleItemsRedered = React.useCallback(
+    ({ visibleRowStartIndex, visibleRowStopIndex }) => {
+      refVisibleIndexes.current = {
+        start: visibleRowStartIndex,
+        stop: visibleRowStopIndex,
+      };
+
+      if (refCurrentPage.current === null) {
+        refCurrentPage.current = 0;
+      }
+
+      if (InfinityScroll) {
+        const { rowsOffset, onReached } = InfinityScroll.props;
+        if (visibleRowStopIndex + rowsOffset > rowCount) {
+          const currentPage = Math.floor(visibleRowStopIndex / pageSize);
+          const nextPage = currentPage + 1;
+          onReached({ currentPage, nextPage });
+        }
+      }
+    },
+    [InfinityScroll, pageSize, rowCount]
+  );
+
+  const handleMouseOver = React.useCallback(
+    event => {
+      highlightRow({ rowIndex: event.target.dataset.rowIndex });
+    },
+    [highlightRow]
+  );
+
+  const handleMouseLeave = React.useCallback(() => {
+    deemphasizeRow();
+  }, [deemphasizeRow]);
+
+  return (
+    <>
+      {hasNoRecords && (
+        <styled.BlockerItem gridId={gridId} $width={gridWidth} $height={height}>
+          <styled.Blocker $width={gridWidth} $height={height}>
+            <div>
```

Comment:
Need this `div`?

Reply:
yes, will let flex to justify and align the content. `(center)`